Userland/Libraries/LibWeb/HTML/StructuredSerialize.cpp

```diff
@@ -936,7 +936,8 @@ class Deserializer {
         auto string_view = TRY(Bindings::throw_dom_exception_if_needed(vm, [&string]() {
             return string->utf8_string_view();
         }));
-        return JS::BigInt::create(vm, ::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1)));
+        auto bigint = MUST(::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1)));
```

Comment:
I have very little knowledge on what the code does, but the word deserialize make me think that this function can take user-provided data (the string check above also kinda confirm that). So we should probably use `TRY` here (probably with that `Bindings::throw_dom_exception_if_needed` that I see for the first time).

Reply:
Looking at #20132 made me think this is unlikely to fail in the normal course of events. I _think_ the correct way to propagate this kind of error would be something like:

```suggestion
        auto bigint_or_error = ::Crypto::SignedBigInteger::from_base(10, string_view.substring_view(0, string_view.length() - 1));
        if (bigint_or_error.is_error())
            return WebIDL::DataCloneError::create(*vm.current_realm(), "Invalid BigInt"_fly_string);
```

I'm not sure if that's the correct thing to do though? I also have very little experience with LibWeb.

Reply:
This shouldn't fail due to invalid formatting of the data, only due to memory allocation failure.

The way that structured data is input into the system is only via the structured_serialize APIs. I don't know of a way currently beyond hacker craziness with a write primitive to get user-controlled data into the stream (unless we have bugs in serialziation/deserialization, e.g. they are not 100% symmetrical).

Reply:
It's ready for a merge then!